mytoyota/models/endpoints/trip.py

```diff
@@ -0,0 +1,95 @@
+from pydantic import BaseModel, Field
+from datetime import date, datetime
+from uuid import UUID
+
+
+class _Pagination(BaseModel):
+    currentPage: int
+    limit: int
+    offset: int
+    pageCount: int
+    previousOffset: int
+    totalCount: int
+
+
+class _SortedBy(BaseModel):
+    field: str
+    order: str
+
+
+class _MetaData(BaseModel):
+    pagination: _Pagination
+    sortedBy: list[_SortedBy]
+
+
+class _Scores(BaseModel):
+    acceleration: int = Field(ge=0, le=100)
+    advice: int = Field(ge=0, le=100, default=0)
+    braking: int = Field(ge=0, le=100)
+    constantSpeed: int = Field(ge=0, le=100, default=0)
+    global_: int | None = Field(ge=0, le=100, alias="global", default=None)
+
+
+class _Summary(BaseModel):
+    countries: list[str]
+    duration: int
+    durationHighway: int
+    durationIdle: int
+    durationOverspeed: int
+    fuelConsumption: float
+    length: int
+    lengthHighway: int
+    lengthOverspeed: int
+    maxSpeed: float
+
+
+class _HDC(BaseModel):
+    # Depending on
+    #    car not being EV
+    #    car being ev only
+    #    car being hybrid but only used ev/fuel
+    chargeDist: int = Field(0)
+    chargeTime: int = Field(0)
+    ecoDist: int = Field(0)
+    ecoTime: int = Field(0)
+    evDistance: int = Field(0)
+    evTime: int = Field(0)
+    powerDist: int = Field(0)
+    powerTime: int = Field(0)
+
+
+class _MonthSummary(BaseModel):
```

Comment:
Is there really only a monthly summary, or also one for days, weeks & years?
If so, should we use a `SummaryBase` class with fields that apply to all time aggregates, from which the `DailySummary`, `MonthlySummary`... classes inherit.

Reply:
There is a daily summary but its in a slightly different format(its part of the histograms) I'd left this for another day didn't want to go to far down the pydantic route before its decided its the way forward.